packages/compass-components/src/components/bson-document-editor.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useRef, useState } from 'react';
+import { css } from '@leafygreen-ui/emotion';
+import { spacing } from '@leafygreen-ui/tokens';
+import type { EditorProps } from './editor';
+import { Editor } from './editor';
+import queryParser from 'mongodb-query-parser';
+import { EJSON } from 'bson';
+import { useElementParentHoverState } from '../utils/use-element-parent-hover-state';
+import { Button } from './leafygreen';
+
+type BSONValue = any;
+
+type BSONDocumentEditorProps = Omit<EditorProps, 'variant'> & {
+  variant?: never;
+  onChangeValue?: (
+    value: BSONValue,
+    error: Error | null,
+    variant: 'EJSON' | 'Shell',
```

Comment:
Logically speaking, we have another way to edit BSON documents, namely the `Document` component from compass-crud. I went back and forth a bit about integrating that one as well, but I don’t think our component structure is currently really laid out for that (e.g. `EditableDocument` itself contains the document actions group, rather than being a component that could be used as an editor straight from the beginning). I’m also not sure if this is something that would make sense from a UX perspective, but ultimately these *are* different representations of the same thing.

Reply:
The editor part of it is in compass-components and separate from the whole card: https://github.com/mongodb-js/compass/blob/main/packages/compass-components/src/components/document-list/document.tsx

There is something there that is sort of an issue: as this was modeled around previous editor, there is no component interface for edits, no `onChange` callback prop, only hadron-document events to subscribe to (but if we wanted to, I think this could be integrated to the current flow)